### PR TITLE
Ingest Rework

### PR DESCRIPTION
**Pull request by minxylynx**
## Description

* Reworked block/transaction ingestion
  * Created procedures to save all block/ tx data at once
    * Should eliminate rogue data insertion
* Migration 1.40 - Data clean up
  * Removed duplicate data for `tx_message`, `tx_msg_event`, `tx_msg_event_attr`
* Migration 1.41 - Added ingestion procedures
  * Added unique indices for all tx-related tables
  * Added types `tx_event`, `tx_msg`, `tx_update`, `block_update`
  * Added procedures `add_tx(tx_update, integer, timestamp)`, `add_block(block_update)`
  * Updated `tx_msg_event_attr` with `attr_idx`, `attr_hash`; updated new columns with data

closes: #274

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/explorer/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by channa-figure — APPROVED** [review]
LGTM

**Review by mtps — APPROVED** [review]

File: service/src/main/kotlin/io/provenance/explorer/domain/models/explorer/ExplorerModels.kt [hunk 1]
```diff
@@ -15,6 +16,14 @@ data class BlockSummary(
     val txNum: Int
 )
 
+data class BlockProposer(
+    var blockHeight: Int,
+    var proposerOperatorAddress: String,
+    var blockTimestamp: DateTime,
+    var minGasFee: Double? = null,
```
Comment (mtps) [inline service/src/main/kotlin/io/provenance/explorer/domain/models/explorer/ExplorerModels.kt:23]: gas fees are typed double? should that be long / biginteger?

**Review by egaxhaj-figure — APPROVED** [review]
LGTM

File: service/src/main/kotlin/io/provenance/explorer/domain/models/explorer/ExplorerModels.kt (hunk 1 above)
Comment (minxylynx) [reply service/src/main/kotlin/io/provenance/explorer/domain/models/explorer/ExplorerModels.kt:23]: This is due to the OG min gas fee amount, which was 0.025 nhash

**Pull request merged**